quesma/quesma/config/config_v2.go

```diff
@@ -0,0 +1,177 @@
+	FrontendConnectors         []FrontendConnector  `koanf:"frontendConnectors"`
+	InstallationId             string               `koanf:"installationId"`
+	LicenseKey                 string               `koanf:"licenseKey"`
+	Logging                    LoggingConfiguration `koanf:"logging"`
+	IngestStatistics           bool                 `koanf:"ingestStatistics"`
+	QuesmaInternalTelemetryUrl *Url                 `koanf:"internalTelemetryUrl"`
+	Processors                 []Processor          `koanf:"processors"`
+}
+
+type FrontendConnector struct {
+	Name   string                         `koanf:"name"`
+	Type   string                         `koanf:"type"`
+	Config FrontendConnectorConfiguration `koanf:"config"`
+}
+
+type FrontendConnectorConfiguration struct {
+	ListenPort network.Port `koanf:"listenPort"`
+}
+
+type BackendConnector struct {
+	Name   string                    `koanf:"name"`
+	Type   string                    `koanf:"type"`
+	Config RelationalDbConfiguration `koanf:"config"`
+}
+
+type Processor struct {
+	Name   string              `koanf:"name"`
+	Type   string              `koanf:"type"`
+	Config QuesmaConfiguration `koanf:"config"`
+}
+
+func LoadV2Config() QuesmaNewConfiguration {
+	var v2config QuesmaNewConfiguration
+	v2config.QuesmaInternalTelemetryUrl = telemetryUrl
+	v2config.Logging.RemoteLogDrainUrl = telemetryUrl
+
+	loadConfigFile()
+	if err := k.Load(env.Provider("QUESMA_", ".", func(s string) string {
+		// This enables overriding config values with environment variables. It's case-sensitive, just like the YAML.
+		// Examples:
+		// `QUESMA_logging_level=debug` overrides `logging.level` in the config file
+		// `QUESMA_licenseKey=arbitrary-license-key` overrides `licenseKey` in the config file
+		return strings.Replace(strings.TrimPrefix(s, "QUESMA_"), "_", ".", -1)
+	}), nil); err != nil {
+		log.Fatalf("error loading config form supplied env vars: %v", err)
+	}
+	if err := k.Unmarshal("", &v2config); err != nil {
+		log.Fatalf("error unmarshalling config: %v", err)
+	}
+
+	return v2config
+}
+
+func (c *QuesmaNewConfiguration) TranslateToLegacyConfig() QuesmaConfiguration {
+	var err, errAcc error
+	var conf QuesmaConfiguration
+	if conf.PublicTcpPort, err = c.getPublicTcpPort(); err != nil {
+		errAcc = multierror.Append(errAcc, err)
+	}
+	if conf.Elasticsearch, err = c.getElasticsearchConfig(); err != nil {
+		errAcc = multierror.Append(errAcc, err)
+	}
+	conf.Logging = c.Logging
+	conf.InstallationId = c.InstallationId
+	conf.LicenseKey = c.LicenseKey
+	conf.IngestStatistics = c.IngestStatistics
+	conf.Connectors = make(map[string]RelationalDbConfiguration)
+	relDBConn, connType, err := c.getRelationalDBConf()
+	if err != nil {
+		errAcc = multierror.Append(errAcc, err)
+	} else {
+		relDBConn.ConnectorType = connType
+		if connType == HydrolixBackendConnectorName {
+			conf.Connectors["injected-hydrolix-connector"] = *relDBConn
```

Comment:
Maybe `injected-hydrolix-connector` and `injected-clickhouse-connector` extract as const or enum

Reply:
I didn't wanted to do that because that's just the "working name" which isn't used/referred anywhere. This will also be gone once we remove the old config object :)